lib/hooks.dart

```diff
@@ -219,3 +219,39 @@ Function useCallback(Function callback, List dependencies) => React.useCallback(
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#usecontext>.
 T useContext<T>(Context<T> context) => ContextHelpers.unjsifyNewContext(React.useContext(context.jsThis));
+
+/// Returns a mutable [Ref] object with [Ref.current] property initialized to [initialValue].
+///
+/// Changes to the [Ref.current] property do not cause the containing [DartFunctionComponent] to re-render.
+///
+/// The returned [Ref] object will persist for the full lifetime of the [DartFunctionComponent].
+/// Compare to [createRef] which returns a new [Ref] object on each render.
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// __Example__:
+///
+/// ```
+/// UseRefTestComponent(Map props) {
+///   final input = useState('');
+///   final inputRef = useRef();
+///   final prevInputRef = useRef();
+///   final prevInput = prevInputRef.current;
+///
+///   useEffect(() {
+///     prevInputRef.current = input.value;
+///   });
+///
+///   return react.Fragment({}, [
+///     react.p({}, ['Current Input: ${input.value}, Previous Input: ${prevInput}']),
+///     react.input({'ref': inputRef}),
+///     react.button({'onClick': (_) => input.set(inputRef.current.value)}, ['Update']),
+///   ]);
+/// }
+/// ```
+///
+/// Learn more: <https://reactjs.org/docs/hooks-reference.html#useref>.
+Ref useRef([dynamic initialValue]) => new Ref.useRefInit(initialValue);
```

Comment:
Ah, I didn't realize this wasn't generic above. I think we could make this generic:
```suggestion
Ref<T> useRef<T>([T initialValue]) => Ref.useRefInit(initialValue);
```

(`Ref.useRefInit` would also have to be made generic instead of just accepting `dynamic`)